### DISCUSSION TITLE
Need more filtering/sorting options for the datasource of a Visualizer

### DISCUSSION THREAD

**Discussion by X3Technology** [DNNCommunity/Dnn.StructuredContent#83] [Ideas]
After selecting a datasource for a visualizer, the only current options are to return everything for a list type visualizer, or manually select a single record for a single type visualizer.  

Should probably also add sorting and paging options (skip/take) as well.

Looking for examples of existing systems that do it well and we can mimic that.

**Comment by X3Technology** [comment]
Anyone have any examples of systems that do it well?

**Comment by david-poindexter** [comment]
Once Razor templates are supported, then this should be easily supported there via normal Razor syntax.  For now, DotLiquid offers some liquid syntax for accomplishing these features.  Is that what you mean?

**Comment by X3Technology** [comment]
Not quite.  I'm basically talking about building the WHERE clause at the database level.  Don't want to return the entire dataset to the front-end to do the filtering/sorting.

And I imagine its possible that the filter parameters will come from a URL querystring or a cookie value, etc.

**Comment by david-poindexter** [comment]
Okay, I see what you are saying.  I would consider this a more advanced feature, but you can certainly do this in 2sxc.
